File: InMotionGIT.Common/MetadataCatalogAttributes/TypeStructureAttribute.vb
Namespace Attributes

    ''' <summary>
    '''
    ''' </summary>
    ''' <remarks></remarks>
    <AttributeUsage(AttributeTargets.Class)>
    Public NotInheritable Class TypeStructureAttribute
        Inherits Attribute

        Private _defaultMember As String

        ''' <summary>
        '''
        ''' </summary>
        Public ReadOnly Property DefaultMember() As String
            Get
                Return _defaultMember
            End Get
        End Property

        Public Sub New(ByVal defaultMember As String)
            _defaultMember = defaultMember
        End Sub

        Public Sub New()
            MyBase.New()
        End Sub

    End Class

End Namespace

File: Common/InMotionGIT.Common.Domain/Attributes/TypeStructureAttribute.cs
using System;

namespace InMotionGIT.Common.Domain.Attributes;


/// <summary>
/// 
/// </summary>
/// <remarks></remarks>
[AttributeUsage(AttributeTargets.Class)]
public sealed class TypeStructureAttribute : Attribute
{

    private string _defaultMember;

    /// <summary>
    /// 
    /// </summary>
    public string DefaultMember
    {
        get
        {
            return _defaultMember;
        }
    }

    public TypeStructureAttribute(string defaultMember)
    {
        _defaultMember = defaultMember;
    }

    public TypeStructureAttribute() : base()
    {
    }

}